shared-libs/server-checks/src/checks.js

```diff
@@ -64,72 +63,54 @@ const couchDbNoAdminPartyModeCheck = () => {
           'see: https://github.com/medic/cht-core/blob/master/DEVELOPMENT.md#enabling-a-secure-couchdb'));
       }
     }).on('error', (e) => {
-      reject(`CouchDB doesn't seem to be running on ${url.format(noAuthUrl)}. ` +
+      reject(`CouchDB doesn't seem to be running on ${noAuthUrl.toString()}. ` +
         `Tried to connect but got an error:\n ${e.stack}`);
     });
   });
 };
 
-const checkNodeName = (nodeName, membership) => {
-  return membership &&
-    membership.all_nodes &&
-    membership.all_nodes.includes(nodeName);
+const getCouchDbVersion = (couchUrl) => {
+  return request.get({ url: couchUrl, json: true }).then(response => response.version);
 };
 
-const couchNodeNamesMatch = (serverUrl) => {
-  const envNodeName = process.env.COUCH_NODE_NAME;
-  const membershipUri = '/_membership';
-  const membershipUrl = serverUrl + membershipUri;
-
-  return new Promise((resolve, reject) => {
-    request.get({ url: membershipUrl, json: true }, (err, response, body) => {
-      if (err) {
-        return reject(err);
-      }
-      if (checkNodeName(envNodeName, body)) {
-        console.log(`Environment variable "COUCH_NODE_NAME" matches server "${envNodeName}"`);
-        return resolve();
-      } else {
-        // we don't want to log user and password, so strip them when we log via getNoAuthURL();
-        const noAuthUrl = getNoAuthURL();
-        noAuthUrl.pathname = membershipUri;
-        return  reject(`Environment variable 'COUCH_NODE_NAME' set to "${envNodeName}" but doesn't match ` +
-          `what's on CouchDB Membership endpoint at "${url.format(noAuthUrl)}". See ` +
-          `https://github.com/medic/cht-core/blob/master/DEVELOPMENT.md#required-environment-variables`);
-      }
-    });
+const couchDbVersionCheck = (couchUrl) => {
+  return getCouchDbVersion(couchUrl).then(version => {
+    console.log(`CouchDB Version: ${version}`);
   });
 };
 
-const getCouchDbVersion = (serverUrl) => {
-  return new Promise((resolve, reject) => {
-    request.get({ url: serverUrl, json: true }, (err, response, body) => {
-      return err ? reject(err) : resolve(body.version);
-    });
-  });
-};
+const couchDbCheck = async (couchUrl) => {
+  let retries = 100;
+  let lastErr;
+  const retryTimeout = () => new Promise(resolve => setTimeout(resolve, 1000));
+  const serverUrl = new URL(couchUrl);
+  serverUrl.pathname = '/';
+
+
+  do {
+    try {
+      retries--;
+      await couchDbVersionCheck(serverUrl.toString());
+      await couchDbNoAdminPartyModeCheck(serverUrl.toString());
+      return;
+    } catch (err) {
+      lastErr = err;
+      console.log('CouchDb check failed', err);
+      await retryTimeout();
+    }
+  } while (retries);
 
-const couchDbVersionCheck = (serverUrl) => {
-  return getCouchDbVersion(serverUrl).then(version => {
-    console.log(`CouchDB Version: ${version}`);
-  });
+  throw lastErr;
```

Comment:
Logging the `err` every attempt is probably a bit scary.

A good old for loop would be more readable I think, eg:

```js
  for (let attempts = 0; attempts < 100; attempts++) {
    try {
      await couchDbVersionCheck(serverUrl.toString());
      await couchDbNoAdminPartyModeCheck(serverUrl.toString());
      return;
    } catch (err) {
      lastErr = err;
      console.log(`CouchDb check ${attempt} failed.`);
      await retryTimeout();
    }
  }
```

Alternatively, maybe we should never give up? If Couch needs some manual intervention wouldn't you expect API to just come up as soon as it's resolved without needing to restart it? This is how sentinel works so maybe just copy that pattern.

Reply:
Updated to keep trying forever.